pallets/lbp/src/tests.rs

```diff
@@ -0,0 +1,1680 @@
+		System::set_block_number(30);
+
+		let user_balance_a_before = Currency::free_balance(ACA, &ALICE);
+		let (balance_a_before, balance_b_before) = LBPPallet::pool_balances(ACA_DOT_POOL_ID);
+
+		assert_noop!(
+			LBPPallet::remove_liquidity(Origin::signed(ALICE), ACA_DOT_POOL_ID, u128::MAX, 0,),
+			Error::<Test>::LiquidityUnderflow
+		);
+
+		let (balance_a_after, balance_b_after) = LBPPallet::pool_balances(ACA_DOT_POOL_ID);
+		assert_eq!(balance_a_after, balance_a_before);
+		assert_eq!(balance_b_after, balance_b_before);
+
+		let user_balance_a_after = Currency::free_balance(ACA, &ALICE);
+		assert_eq!(user_balance_a_after, user_balance_a_before);
+
+		expect_events(vec![
+			Event::PoolCreated(ALICE, ACA, DOT, 1_000_000_000, 2_000_000_000).into(),
+			Event::Unpaused(ALICE, ACA_DOT_POOL_ID).into(),
+		]);
+	});
+}
+
+#[test]
+fn remove_liquidity_during_sale_should_not_work() {
+	predefined_test_ext().execute_with(|| {
+		// sale started at the block number 10
+		System::set_block_number(15);
+
+		let user_balance_a_before = Currency::free_balance(ACA, &ALICE);
+		let (balance_a_before, balance_b_before) = LBPPallet::pool_balances(ACA_DOT_POOL_ID);
+
+		assert_noop!(
+			LBPPallet::remove_liquidity(Origin::signed(ALICE), ACA_DOT_POOL_ID, 1_000, 0,),
+			Error::<Test>::SaleNotEnded
+		);
+
+		let (balance_a_after, balance_b_after) = LBPPallet::pool_balances(ACA_DOT_POOL_ID);
+		assert_eq!(balance_a_after, balance_a_before);
+		assert_eq!(balance_b_after, balance_b_before);
+
+		let user_balance_a_after = Currency::free_balance(ACA, &ALICE);
+		assert_eq!(user_balance_a_after, user_balance_a_before);
+
+		expect_events(vec![
+			Event::PoolCreated(ALICE, ACA, DOT, 1_000_000_000, 2_000_000_000).into(),
+			Event::Unpaused(ALICE, ACA_DOT_POOL_ID).into(),
+		]);
+	});
+}
+
+#[test]
+fn destroy_pool_should_work() {
+	predefined_test_ext().execute_with(|| {
+		System::set_block_number(21);
+
+		let user_balance_a_before = Currency::free_balance(ACA, &ALICE);
+		let user_balance_b_before = Currency::free_balance(DOT, &ALICE);
+		let user_balance_hdx_before = Currency::reserved_balance(HDX, &ALICE);
+		let (balance_a_before, balance_b_before) = LBPPallet::pool_balances(ACA_DOT_POOL_ID);
+
+		assert_ok!(LBPPallet::destroy_pool(Origin::signed(ALICE), ACA_DOT_POOL_ID,));
+
+		let (balance_a_after, balance_b_after) = LBPPallet::pool_balances(ACA_DOT_POOL_ID);
+		assert_eq!(balance_a_after, 0);
+		assert_eq!(balance_b_after, 0);
+
+		let user_balance_a_after = Currency::free_balance(ACA, &ALICE);
+		assert_eq!(
+			user_balance_a_after,
+			user_balance_a_before.saturating_add(balance_a_before)
+		);
+
+		let user_balance_b_after = Currency::free_balance(DOT, &ALICE);
+		assert_eq!(
+			user_balance_b_after,
+			user_balance_b_before.saturating_add(balance_b_before)
+		);
+
+		let user_balance_hdx_after = Currency::reserved_balance(HDX, &ALICE);
+		assert_eq!(
+			user_balance_hdx_after,
+			user_balance_hdx_before.saturating_sub(POOL_DEPOSIT)
+		);
+
```

Comment:
I would also add check that storage was cleared ( assets, pool data, owner, balances, deposit).